File: Java/src/main/java/com/backend/java/model/PodcastContent.java
package com.backend.java.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonTypeName;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;

@Entity
@Table(name = "podcast_content")
@JsonTypeName("podcast")
public class PodcastContent extends Content {
    @JsonProperty("audioUrl")
    private String audioUrl;
    @JsonProperty("duration")
    private int duration;

    // Constructors, getters, and setters
    public PodcastContent() {}

    public PodcastContent(String audioUrl, int duration) {
        this.audioUrl = audioUrl;
        this.duration = duration;
    }

    public String getAudioUrl() {
        return audioUrl;
    }

    public void setAudioUrl(String audioUrl) {
        this.audioUrl = audioUrl;
    }

    public int getDuration() {
        return duration;
    }

    public void setDuration(int duration) {
        this.duration = duration;
    }

    @Override
    public String toString() {
        return "PodcastContent{" +
                "audioUrl='" + audioUrl + '\'' +
                '}';
    }
}

File: Csharp/Models/PodcastContent.cs
namespace Models
{
    public class PodcastContent : Content
    {
        public string AudioUrl { get; set; }
        public int Duration { get; set; }

        public PodcastContent() 
        {
            Type = "podcast";
        }

        public PodcastContent(string audioUrl, int duration)
        {
            AudioUrl = audioUrl;
            Duration = duration;
            Type = "podcast"; 
        }
    }
}